### ISSUE TITLE
Improving custom object detection false positives

### ISSUE THREAD

**Issue by dhasl002** [tensorflow/models#7636]
### System information
- **What is the top-level directory of the model you are using**: research/object_detection/
- **Have I written custom code**: No, I followed the [tutorial](https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/training.html)
- **OS Platform and Distribution**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.14.0
- **Bazel version **: N/A
- **CUDA/cuDNN version**: 10.0
- **GPU model and memory**:  NVIDIA Tesla V100 
- **Exact command to reproduce**: N/A

### Describe the problem
Feature Recommendation:
I am using the tensorflow custom detection api to detect custom furniture. I have found that I get very good accuracy when I see one of the classes. However, when seeing images that are not of any category, there is a high likelihood of seeing false positives. 

I googled around looking for a solution ([issue 29078](https://github.com/tensorflow/tensorflow/issues/29078), [SO post](https://stackoverflow.com/questions/49687809/tensorflow-high-false-positive-rate-and-non-max-suppression-issue), #2544). From my research, I could not find a definitive answer.

I was able to solve this problem myself and think that a feature could solve this problem for others. My solution was to randomly feed in unlabeled images from the coco dataset to show my model negative examples. I believe that this could be a data augmentation option. 

How do others feel about this? Thanks!

### Source code / logs
N/A

**Comment by Tanmay-Kulkarni101** [comment]
It seems like a great idea. To date, I used to make use of other objects which are found in the problem domain, without any labels, to be used as negative examples. Making use of the coco dataset may reduce programming effort. However, it begs the question, how to you find examples that relate to the particular domain, that you are trying to predict ?

**Comment by FIL228** [comment]
I don't know nothing about all of that and just buy  in Wish a material to make a central météo in kit and just want all to made that. Envoyé depuis mon mobile Huawei-------- Message original --------Objet : Re: [tensorflow/models] Improving custom object detection false positives (#7636)De : Tanmay Kulkarni À : tensorflow/models Cc : Subscribed It seems like a great idea. To date, I used to make use of other objects which are found in the problem domain, without any labels, to be used as negative examples. Making use of the coco dataset may reduce programming effort. However, it begs the question, how to you find examples that relate to the particular domain, that you are trying to predict ?

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/tensorflow/models/issues/7636?email_source=notifications\u0026email_token=ANZAZGGUU4T3HOJOU5BEHTLQT3RE3A5CNFSM4I6VHMK2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEGILCQ#issuecomment-554468746",
"url": "https://github.com/tensorflow/models/issues/7636?email_source=notifications\u0026email_token=ANZAZGGUU4T3HOJOU5BEHTLQT3RE3A5CNFSM4I6VHMK2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEGILCQ#issuecomment-554468746",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by sampathkumaran90** [comment]
I am facing similar issue. When i set min_negatives_per_image to 0 and not set any negative images, i get prediction correct for positive images but getting a lot of false positives. When i add a bunch of negative images say 50% positive and 50% negative and set min_negatives_per_image to 10, the results are not good. I am getting more false negatives as well as false positives. Anyone could help me on this issue... I am using ssd mobilenet v2 or finetuning
